resources/benchmark-runner.mjs

```diff
@@ -41,43 +41,132 @@ class BenchmarkState {
         return !this._testIndex;
     }
 
-    prepareCurrentSuite(runner, frame) {
+    async prepareCurrentSuite(frame, page) {
         const suite = this.currentSuite();
-        return new Promise((resolve) => {
-            frame.onload = () => {
-                suite.prepare(runner, frame.contentWindow, frame.contentDocument).then(resolve);
+        await new Promise((resolve) => {
+            frame.onload = async () => {
+                await suite.prepare(page);
+                resolve();
             }
             frame.src = 'resources/' + suite.url;
         });
     }
 }
 
-export class BenchmarkRunner {
-    constructor(suites, client) {
-        this._suites = suites;
-        this._prepareReturnValue = null;
-        this._client = client;
+class Page {
+    constructor(frame) {
+        this._frame = frame;
     }
 
-    waitForElement(selector) {
+    async waitForElement(selector) {
         return new Promise((resolve) => {
-            const contentDocument = this._frame.contentDocument;
-
-            function resolveIfReady() {
-                const element = contentDocument.querySelector(selector);
-                if (element) {
-                    window.requestAnimationFrame(function () {
-                        return resolve(element);
-                    });
-                    return;
-                }
+          const resolveIfReady = () => {
+            const element = this.querySelector(selector);
+            if (element) {
+                window.requestAnimationFrame(() => {
+                    return resolve(element);
+                });
+            } else {
                 setTimeout(resolveIfReady, 50);
             }
-
-            resolveIfReady();
+          };
+          resolveIfReady();
         });
     }
 
+    querySelector(selector) {
+        const element = this._frame.contentDocument.querySelector(selector);
+        if (element === null) return null
+        return this._wrapElement(element);
+    }
+
+    querySelectorAll(selector) {
+        const elements = Array.from(this._frame.contentDocument.querySelectorAll(selector));
+        for (let i = 0; i < elements.length; i++) {
+            elements[i] = this._wrapElement(elements[i]);
+        }
+        return elements
+    }
+
+    getElementById(id) {
+        const element = this._frame.contentDocument.getElementById(id);
+        if (element === null) return null
+        return this._wrapElement(element)
+    }
+
+    eval(source) {
+        this._frame.contentWindow.eval(source);
+        return null;
+    }
+
+    _wrapElement(element) {
+        return new PageElement(element);
+    }
+}
+
+const ENTER_KEY_CODE = 13;
+
+const NATIVE_OPTIONS = {
+    bubbles: true,
+    cancellable: true,
+}
+
+class PageElement {
+    constructor(node) {
+        this._node = node
+    }
+
+    set value(value) {
```

Comment:
We typically prefer having a named method instead of using setter/getter in these cases.